feat(cli): add /chat file commands for session management (#3190)
QwenLM/qwen-code: .qwen/commands/chat-resume.md

# chat-resume.md — Resume a Saved Session

1. Validate `{{name}}` (Common rules): `^[a-zA-Z0-9_.-]+$`, ≤128, ≠ `.`/`..`/`__proto__`/`constructor`/`prototype`.
2. Look up ID in index (`.qwen/chat-index.json` in project root, NOT `~/.qwen/`). Missing/not found → show list + "Session not found", stop.
3. Verify `~/.qwen/projects/<sanitizeCwd>/chats/<id>.jsonl` exists. Missing → warn "Session file missing", stop.
4. **Execute a shell command** to launch a NEW terminal window. Run the command below using your shell tool. **DO NOT read the .jsonl file content.**

>    - Windows: run `start pwsh -NoExit -Command "qwen --resume <id>"`. If it fails, run `start cmd /k "qwen --resume <id>"`

Reviewer: **[Critical]** `chat-resume` validates only the saved name, then reads `<id>` from `.qwen/chat-index.json` and interpolates it directly into shell commands such as `qwen --resume <id>` / `osascript ... "qwen --resume <id>"` without validating or shell-escaping the stored ID. A malformed local index entry can turn `/chat -r` into command execution if a matching crafted `.jsonl` file exists. Please validate the loaded ID against the expected session filename format, reject anything else, and quote/escape it in every platform command.

_— gpt-5.5 via Qwen Code /review_

# chat-resume.md — Resume a Saved Session

> 1. Validate `{{name}}` (Common rules): `^[a-zA-Z0-9_.-]+$`, ≤128, ≠ `.`/`..`/`__proto__`/`constructor`/`prototype`.

Reviewer: **[Critical]** This validation does not match the real session ID format accepted by the CLI. `qwen --resume` accepts standard hyphenated UUIDs, optionally followed by `-agent-...`, but this rule accepts `32 hex chars or 8-char prefix`. A session saved from a normal `.jsonl` filename can therefore be rejected, while an 8-character prefix can pass this command spec and then fail or resolve incorrectly later. Please align this with the CLI's full session ID format, or resolve prefixes to a unique full session ID before invoking `qwen --resume`.

_— gpt-5.5 via Qwen Code /review_

> 2. Look up ID in index (`.qwen/chat-index.json` in project root, NOT `~/.qwen/`). Missing/not found → show list + "Session not found", stop.

Reviewer: **[Critical]** Checking only that `<id>.jsonl` exists in the sanitized project directory is not enough to prove the session belongs to the current project. Core `SessionService` reads the first record and compares the recorded project identity because sanitized project directory names can collide. Without the same check, this command can resume a session from another project that shares the sanitized directory. Please verify the first JSONL record belongs to the current project before accepting the saved ID.

_— gpt-5.5 via Qwen Code /review_

> 3. **Validate loaded ID**: The ID from index must match the expected UUID format (32 hex chars or 8-char prefix). If ID contains any shell metacharacters (`$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces), reject it: "Error: Invalid session ID from index. Aborted." — **DO NOT execute any shell command with this ID**.

Reviewer: **[Critical]** The new terminal commands launch `qwen --resume <id>` without preserving the current project directory. Named sessions are project-scoped, and a new terminal can start in the user's home/default directory, especially with the macOS `osascript` path, so the command may verify a session in the current project and then launch Qwen in a different cwd where that session cannot be found. Please include an escaped `cd <projectRoot> && qwen --resume <id>` equivalent for each platform.

_— gpt-5.5 via Qwen Code /review_

# chat-resume.md — Resume a Saved Session

1. Validate `{{name}}` (Common rules): `^[a-zA-Z0-9_.-]+$`, ≤128, ≠ `.`/`..`/`__proto__`/`constructor`/`prototype`.
2. Look up ID in index (`.qwen/chat-index.json` in project root, NOT runtime base). Missing/not found → show list + "Session not found", stop.

> 3. **Validate loaded ID**: The ID from index must match UUID format (`^[a-fA-F0-9-]+$`, allows hyphens). If ID contains any shell metacharacters (`$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces), reject it: "Error: Invalid session ID from index. Aborted." — **DO NOT execute any shell command with this ID**.

Reviewer: **[Suggestion]** Session-ID validation regex `^[a-fA-F0-9-]+$` accepts strings starting with `-` (e.g., `-rf`, `--help`, `-version`). Such an ID becomes an **option flag** when interpolated into `qwen --resume <id>` — argv-injection (not shell injection) into the spawned `qwen` process.

Real session UUIDs don't start with `-`, but `chat-index.json` is plain JSON in the project root and `chat-save.md` explicitly invites manual editing ("Fix it manually before saving."). A malicious or accidentally hand-edited index value like `"my-session": "--help"` would cause `qwen --resume --help` instead of resuming. Future `qwen` flags accepting values (e.g., a hypothetical `--config <file>`) would let an attacker who can write to `.qwen/chat-index.json` redirect program behavior on the victim's next `/chat -r`.

Suggested fix — tighten the validator to a real UUID shape:

```
The ID from index must match canonical UUID format:
  ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
(with optional `-agent-...` suffix matching `(-agent-[a-zA-Z0-9_-]+)?` if needed).
```

At minimum: reject any ID starting with `-`.

_via Qwen Code /review_

> 4. **Verify session belongs to current project**: Read the first line of `<runtimeBase>/projects/<sanitizeCwd>/chats/<id>.jsonl`. Parse JSON and verify `project` field matches the current project directory. If mismatch → "Error: Session belongs to another project. Aborted." Missing file → warn "Session file missing", stop.

Reviewer: **[Critical]** Step 4 says "Parse JSON and verify `project` field matches the current project directory." The actual qwen-code session record has **no `project` field** — see `packages/core/src/services/sessionService.ts:592` which compares `getProjectHash(firstRecord.cwd) === this.projectHash`. The real field is `cwd`, and the real comparison hashes both sides via `getProjectHash` (SHA-256 of sanitized path).

With the field absent, an LLM following the spec literally will either always reject (`undefined === <projectDir>` → false → "Session belongs to another project. Aborted.") or silently skip the check; either way the safety check the prior review asked for is dead code. The ADDRESSED count above is one fewer than it appears.

Additional gaps in the same step (relevant once the field name is fixed):

- **Empty file** (zero bytes — common during interrupted writes): no defined behavior. Should be "Session file empty (likely interrupted save). Aborted."
- **First record is not valid JSON** (truncated mid-record from a crash): no defined behavior.
- **First record is a system message lacking `cwd`**: no defined behavior.

Suggested fix:

```
4. **Verify session belongs to current project**: Read the first JSON line of `<runtimeBase>/projects/<sanitizeCwd>/chats/<id>.jsonl`. 
   - File missing → "Session file missing", stop.
   - File 0 bytes → "Session file empty (likely interrupted save). Aborted."
   - First line not valid JSON → "Session file corrupt at line 1. Aborted."
   - JSON has no `cwd` field → "Session record missing project context. Aborted."
   - JSON `cwd` field, after running through the same sanitizeCwd as this command, ≠ current project's `<sanitizeCwd>` → "Session belongs to another project. Aborted."
```

Alternatively (simpler), drop the in-LLM verification entirely and rely on the path itself: the file is at `<runtimeBase>/projects/<sanitizeCwd>/chats/<id>.jsonl`, so the path already encodes the project bucket. If `<id>.jsonl` exists at the current project's `<sanitizeCwd>` directory, it belongs to this project by construction.

_via Qwen Code /review_

5. Verify `<runtimeBase>/projects/<sanitizeCwd>/chats/<id>.jsonl` exists. Missing → warn "Session file missing", stop.
6. **Execute a shell command** to launch a NEW terminal window with cd to project directory first:

>    - Windows: `start pwsh -NoExit -Command "cd '<projectRoot>'; qwen --resume <escaped_id>"` (escape `<id>` by replacing `"` with `\"`)

Reviewer: **[Critical]** `<projectRoot>` is referenced 3× in Step 6 (Windows / macOS / Linux) but **never defined or sourced** in any of the .md files. The LLM has to guess where it comes from:

- Current `qwen` process cwd? Tautological — the new terminal would be opening in the same directory the user is already in.
- Decoded from `<sanitizeCwd>`? Impossible — `sanitizeCwd()` is not invertible (`D:\code\qwen-code` → `d--code-qwen-code`, can't recover the original).
- The `cwd` field of the first record in `<id>.jsonl`? Correct, but the spec never says so.

Suggested fix — add Step 4.5 (or fold into Step 4 once the `cwd`/`project` issue is fixed):

```
4.5. Set <projectRoot> = the cwd field of the first JSON record in <id>.jsonl. Verify it exists on disk. Missing → "Error: original project directory <projectRoot> no longer exists. Aborted."
```

This also enables a useful safety check: if the saved session was originally in a directory that's been deleted/moved, the user gets a clean error instead of a terminal that flickers and dies when `cd` fails silently.

_via Qwen Code /review_

>    - macOS: `osascript -e 'tell app "Terminal" to do script "cd '<projectRoot>'; qwen --resume <escaped_id>"'` (escape `<id>` by replacing `"` with `\"`)

Reviewer: **[Critical]** macOS osascript line is malformed regardless of `<projectRoot>` content:

```
osascript -e 'tell app "Terminal" to do script "cd '<projectRoot>'; qwen --resume <escaped_id>"'
```

POSIX shells **do not nest single quotes**. The first `'` immediately after `cd ` terminates the outer `-e` argument, splices in unquoted `<projectRoot>` (so spaces become argv splits), then reopens. As literally templated, this command produces a shell parse / osascript syntax error or — worse — silent path truncation on macOS hosts where `$HOME` contains a space (the default `/Users/Alice Smith/...` pattern with first+last name accounts).

A literal-following LLM cannot produce a working command from this spec.

Suggested fix — use double-quote outer with explicit escaping inside, and POSIX-shell-quote `<projectRoot>` (single-quote wrapped, internal `'` rewritten as `'\''`):

```
osascript -e "tell app \"Terminal\" to do script \"cd <projectRootEscaped> && qwen --resume <id>\""
```

Where `<projectRootEscaped>` is the result of replacing every `'` in `<projectRoot>` with `'\''` and wrapping the whole string in `'…'`. Apply the same treatment to the Linux line (`bash -c '…'`) and the Windows line (PowerShell single-quote inside `"…"`).

Alternatively: write the AppleScript to a temp file (`mktemp /tmp/qwen-resume.XXXXXX.scpt`) and run `osascript /tmp/qwen-resume.XXXXXX.scpt` — sidesteps shell quoting entirely.

_via Qwen Code /review_

2. Look up ID in index (`.qwen/chat-index.json` in project root, NOT runtime base). Missing/not found → show list + "Session not found", stop.
3. **Validate loaded ID**: The ID from index must match UUID format (`^[a-fA-F0-9-]+$`, allows hyphens). If ID contains any shell metacharacters (`$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces), reject it: "Error: Invalid session ID from index. Aborted." — **DO NOT execute any shell command with this ID**.
4. **Get session project directory**: Read the first line of `<runtimeBase>/projects/<sanitizeCwd>/chats/<id>.jsonl`.
   - File missing → "Session file missing", stop.
   - File 0 bytes → "Session file empty (likely interrupted save). Aborted.", stop.
   - First line not valid JSON → "Session file corrupt at line 1. Aborted.", stop.
   - JSON has no `cwd` field → "Session record missing project context. Aborted.", stop.
   - Set `<projectRoot>` = the `cwd` field value from the JSON record.
   - Verify `<projectRoot>` directory exists on disk. Missing → "Error: original project directory '<projectRoot>' no longer exists. Aborted.", stop.

> 5. **Verify session belongs to current project**: Apply `sanitizeCwd(<projectRoot>)` and compare with current project's `<sanitizeCwd>`. If they don't match → "Error: Session belongs to another project. Aborted.", stop.

Reviewer: **[Suggestion]** Project-identity verification uses `sanitizeCwd()` comparison, but the core `SessionService` uses `getProjectHash()` (SHA-256) for all project-ownership checks. `sanitizeCwd` is not collision-resistant — two different paths can produce the same sanitized form (e.g., `/home/a-b/c` and `/home/a/b-c` both become `-home-a-b-c`).

Consider computing SHA-256 via `node -e "require('crypto').createHash('sha256').update(process.argv[1]).digest('hex')" '<projectRoot>'` to match the core's approach, or document this as a known limitation of file-based commands.

_— glm-5.1 via Qwen Code /review_

> 6. **Execute a shell command** to launch a NEW terminal window with cd to project directory:

Reviewer: **[Critical]** `<projectRoot>` is sourced from the JSONL `cwd` field (step 4) and interpolated into shell commands without shell-safety validation. Two concrete problems:

1. **POSIX platforms** (macOS line 17, Linux line 19): `cd '<projectRoot>'` — a path containing a single quote (e.g., `/Users/O'Brien/project`) breaks out of single-quote wrapping.
2. **Windows CMD** (line 16): `cd /d <projectRoot>` — completely unquoted, so paths with spaces (e.g., `C:\Program Files\project`) cause command splitting.

The session ID `<id>` is properly validated (`^[a-fA-F0-9-]+$`), but `<projectRoot>` has no equivalent check. Consider adding a validation step that rejects `<projectRoot>` if it contains shell metacharacters, or documents per-platform escaping rules.

_— glm-5.1 via Qwen Code /review_

   - Windows (PowerShell): `start pwsh -NoExit -Command "cd '<projectRoot>'; qwen --resume <id>"`
   - Windows (CMD fallback): `start cmd /k "cd /d <projectRoot> && qwen --resume <id>"` (use if PowerShell unavailable)
   - macOS: `osascript -e "tell app \"Terminal\" to do script \"cd '<projectRoot>' && qwen --resume <id>\""`

>    - Linux (WSL): If platform is linux and `/proc/version` contains "Microsoft" or "WSL", use: `cmd.exe /c "start cmd /k cd /d <projectRoot> && qwen --resume <id>"` or prefer `wt.exe` if available

Reviewer: **[Critical]** WSL resume command passes a Linux-style `<projectRoot>` (e.g., `/home/user/project`) to `cmd.exe /c cd /d`, which expects a Windows path. The `cwd` field from the JSONL session file stores the Linux-native path. `cd /d` will fail with "The system cannot find the path specified." Additionally, `qwen` is typically installed within the WSL distribution and not in Windows CMD's PATH.

Consider using `wslpath -w "<projectRoot>"` to convert to a Windows path, or launching within the WSL context (e.g., `wsl.exe -e bash -c "cd '<projectRoot>' && qwen --resume <id>"`).

_— glm-5.1 via Qwen Code /review_

>    - Linux (native): detect terminal with `command -v` (gnome-terminal, xterm, alacritty, kitty in order), then run: `<terminal> -- bash -c "cd '<projectRoot>' && qwen --resume <id>'"`

Reviewer: **[Critical]** Linux native command has a stray trailing single quote. `<terminal> -- bash -c "cd '<projectRoot>' && qwen --resume <id>'"` — the `'` after `<id>` is unmatched, causing `bash` to hang indefinitely waiting for input with no error output. The source file (`chat-src/commands/chat-resume.md`) correctly omits this stray quote.

```suggestion
   - Linux (native): detect terminal with `command -v` (gnome-terminal, xterm, alacritty, kitty in order), then run: `<terminal> -- bash -c "cd '<projectRoot>' && qwen --resume <id>"`
```

_— glm-5.1 via Qwen Code /review_

# chat-resume.md — Resume a Saved Session

1. Validate `{{name}}` (Common rules): `^[a-zA-Z0-9_.-]+$`, ≤128, ≠ `.`/`..`/`__proto__`/`constructor`/`prototype`.

> 2. Look up ID in index (`.qwen/chat-index.json` in project root, NOT runtime base). Missing/not found → show list + "Session not found", stop.

Reviewer: **[Suggestion]** chat-resume is the only sub-command missing malformed JSON handling for `chat-index.json`. The other three all specify: `"JSON parse error → output 'chat-index.json is malformed. Fix it manually.' and stop."` If the index file is corrupted, chat-resume behaves unpredictably instead of showing a clear error.

```suggestion
2. Look up ID in index (`.qwen/chat-index.json` in project root, NOT runtime base). **JSON parse error → output `"chat-index.json is malformed. Fix it manually before resuming."` and stop. Do NOT proceed.** Missing/not found → show list + "Session not found", stop.
```

3. **Validate loaded ID**: The ID from index must match UUID format (`^[a-fA-F0-9-]+$`, allows hyphens). If ID contains any shell metacharacters (`$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces), reject it: "Error: Invalid session ID from index. Aborted." — **DO NOT execute any shell command with this ID**.
4. **Get session project directory**: Read the first line of `<runtimeBase>/projects/<sanitizeCwd>/chats/<id>.jsonl`.
   - File missing → "Session file missing", stop.
   - File 0 bytes → "Session file empty (likely interrupted save). Aborted.", stop.
   - First line not valid JSON → "Session file corrupt at line 1. Aborted.", stop.
   - JSON has no `cwd` field → "Session record missing project context. Aborted.", stop.
   - Set `<projectRoot>` = the `cwd` field value from the JSON record.
   - Verify `<projectRoot>` directory exists on disk. Missing → "Error: original project directory '<projectRoot>' no longer exists. Aborted.", stop.
5. **Verify session belongs to current project**: Apply `sanitizeCwd(<projectRoot>)` and compare with current project's `<sanitizeCwd>`. If they don't match → "Error: Session belongs to another project. Aborted.", stop.

>    **Limitation note**: chat-resume uses sanitizeCwd for project comparison. The core SessionService uses SHA-256 hash (getProjectHash). sanitizeCwd may have collisions for unusual paths, but this is a known limitation of file-based commands.

Reviewer: **[Suggestion]** The limitation note states "The core SessionService uses SHA-256 hash (getProjectHash)" — but `Storage.getProjectDir()` (which determines the actual session storage path) uses `sanitizeCwd`, not `getProjectHash`. Both the chat commands and the core code use the same function; there is no architectural mismatch.

This factual error could mislead a future contributor into believing there's a real gap between layers and wasting time "fixing" something that isn't broken.

```suggestion
   **Limitation note**: chat-resume uses sanitizeCwd for project comparison. Both these commands and the core SessionService use `sanitizeCwd` for session directory resolution. The collision risk (e.g., `/home/a-b/c` and `/home/a/b-c` both produce `home-a-b-c`) is inherent in the sanitizeCwd algorithm itself, not a mismatch between layers.
```

> 6. **Validate projectRoot for shell safety**: If <projectRoot> contains any shell metacharacters (`$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces), reject: "Error: Session path contains unsafe characters. Aborted."

Reviewer: **[Critical]** Step 6 shell-safety validation rejects `$` `` ` `` `;` `|` `>` `<` `&` `(` `)` and spaces, but does **not** reject single quotes (`'`). However, step 8's Linux command wraps `<projectRoot>` in single quotes (`cd '<projectRoot>'`), and the macOS command also relies on single-quote wrapping. A path containing a single quote (e.g., `/Users/O'Brien/project`) will prematurely terminate the single-quoted string, producing a broken or potentially dangerous shell command.

**Suggested fix:** Add `'` (single quote / apostrophe) to the POSIX metacharacter rejection list in step 6.

```suggestion
6. **Validate `projectRoot` for Shell Safety**: Check `<projectRoot>` for metacharacters. If it contains any of `$` ` ` `;` `|` `>` `<` `&` `(` `)` `'` (or spaces — see note), output "Error: Session path contains unsafe characters. Aborted." and stop. For Windows: also check for `^` and `%`.
```

_— pai/glm-5.1 via Qwen Code /review_

> 6. **Validate projectRoot for shell safety**: If <projectRoot> contains any shell metacharacters (`$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces), reject: "Error: Session path contains unsafe characters. Aborted."

Reviewer: **[Critical]** Step 6 rejects paths containing spaces, but all platform command templates already properly quote the path (PowerShell: `cd '<projectRoot>'`, CMD: `cd /d \"<projectRoot>\"`, macOS/Linux: `cd '<projectRoot>'`). Spaces are not a shell injection vector — they are a normal path character handled by quoting. Rejecting spaces makes `/chat resume` unusable for projects with spaces in their directory path (e.g., `~/My Project/`, `C:\Program Files\project`), which are extremely common on macOS and Windows.

**Suggested fix:** Remove spaces from the POSIX metacharacter rejection list. Keep only truly dangerous shell metacharacters (`$` `` ` `` `;` `|` `>` `<` `&` `(` `)` `'`):

```suggestion
6. **Validate `projectRoot` for Shell Safety**: Check `<projectRoot>` for metacharacters. If it contains any of `$` ` ` `;` `|` `>` `<` `&` `(` `)` `'`, output "Error: Session path contains unsafe characters. Aborted." and stop. For Windows: also check for `^` and `%`.
```

_— pai/glm-5.1 via Qwen Code /review_

> 6. **Validate projectRoot for shell safety**: If <projectRoot> contains any shell metacharacters (`$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces), reject: "Error: Session path contains unsafe characters. Aborted."

Reviewer: **[Critical]** projectRoot shell metacharacter blocklist is incomplete — allows arbitrary command execution

Step 6 blocks `$`, `` ` ``, `;`, `|`, `>`, `<`, `&`, `(`, `)`, spaces. But **newline (`\n`), double-quote (`"`), single-quote (`'`), and backslash (`\`)** are missing.

Three concrete injection vectors:
- **Linux**: `\n` inside `bash -c "cd '<projectRoot>' && ..."` acts as a command separator — a newline in the path executes arbitrary commands
- **macOS**: `"` in `echo "<projectRoot>"` closes the double-quote context, allowing injection via the osascript command
- **All POSIX**: `'` breaks `cd '<projectRoot>'` single-quote delimiters — e.g., `/x'; touch /tmp/pwned; echo '/x` injects `touch /tmp/pwned`

Attack chain: a crafted `.jsonl` file with a malicious `cwd` field in `~/.qwen/projects/<sanitizeCwd>/chats/` enables RCE when the user runs `/chat -r`.

```suggestion
6. **Validate projectRoot for shell safety**: <projectRoot> must match `^[a-zA-Z0-9/._-]+$` — reject any path containing characters outside this set. Reject: "Error: Session path contains unsafe characters. Aborted."
   - For Windows: also reject `^`, `%`, `\`
```

   - For Windows: also check for `^` and `%`
7. **Execute a shell command** to launch a NEW terminal window with cd to project directory:
   - Windows (PowerShell): `start pwsh -NoExit -Command "cd '<projectRoot>'; qwen --resume <id>"`
   - Windows (CMD fallback): `start cmd /k "cd /d \"<projectRoot>\" && qwen --resume <id>"` (use if PowerShell unavailable)

>    - macOS: `osascript -e "tell app \"Terminal\" to do script \"cd '$(echo "<projectRoot>" | sed "s/'/\\\\'/g")' && qwen --resume <id>\""`

Reviewer: **[Critical]** The macOS `sed` escaping `sed "s/'/\\\\'/g"` produces `\'` as the replacement for `'`. However, inside a bash single-quoted string, `\'` is **not** a valid escape — backslash has no special meaning inside single quotes, so `'` still terminates the string. The correct bash idiom for embedding a single quote inside a single-quoted string is `'\''` (end-quote, escaped-quote, reopen-quote).

If Finding 1 is adopted (adding `'` to the step 6 rejection list), this sed escaping becomes unnecessary and can be removed. If single-quote paths must be supported, the sed replacement must produce `'\\''` instead of `\'`:

```suggestion
cd '$(echo "<projectRoot>" | sed "s/'/'\\\\''/g")' && qwen --resume <id>
```

_— pai/glm-5.1 via Qwen Code /review_

   - Linux (WSL): If platform is linux and `/proc/version` contains "Microsoft" or "WSL":
     - Convert path: Run `wslpath -w "<projectRoot>"` to get Windows path

>      - Use: `cmd.exe /c "start cmd /k cd /d \"<windowsPath>\" && qwen --resume <id>"` or prefer `wt.exe -d "<windowsPath>" -- qwen.exe --resume <id>"`

Reviewer: **[Critical]** The WSL `wt.exe` command template has a stray trailing double quote: `wt.exe -d "<windowsPath>" -- qwen.exe --resume <id>"`. The `"` after `<id>` is unmatched and will be appended to the session ID argument, causing `qwen.exe --resume` to receive an incorrect ID and fail.

**Suggested fix:** Remove the trailing `"`:

```suggestion
wt.exe -d "<windowsPath>" -- qwen.exe --resume <id>
```

_— pai/glm-5.1 via Qwen Code /review_

1. Validate `{{name}}` (Common rules).
2. Look up ID in index. Missing → show list, stop.
3. Verify `~/.qwen/projects/<hash>/chats/<id>.jsonl` exists. Missing → warn, stop.
4. Open new window (detect OS):

>    - Windows: `start pwsh -NoExit -Command "qwen --resume <id>"`

Reviewer: **[Critical]** Production file omits Windows CMD fallback (source has both `start pwsh ...` and `start cmd /k ...`). Also Linux fallback has literal `...` instead of full command `xterm -e "qwen --resume <id>"`. Windows systems without PowerShell can't resume; the `...` is ambiguous for the LLM.

```suggestion
   - Windows: `start pwsh -NoExit -Command "qwen --resume <id>"` or `start cmd /k "qwen --resume <id>"`
   - macOS: `osascript -e 'tell app "Terminal" to do script "qwen --resume <id>"'`
   - Linux: `gnome-terminal -- qwen --resume <id>` (or `xterm -e "qwen --resume <id>"`)
```

_— qwen3.6-plus via Qwen Code /review_

   - macOS: `osascript -e 'tell app "Terminal" to do script "qwen --resume <id>"'`

>    - Linux: `gnome-terminal -- qwen --resume <id>` (or `xterm -e ...`)

Reviewer: **[Suggestion]** Linux terminal detection hardcodes `gnome-terminal` with no actual fallback detection. On non-GNOME Linux (KDE, Sway, WSL, containers), resume will fail. Add a `command -v` check:

```suggestion
   - Linux: Run `command -v gnome-terminal && gnome-terminal -- qwen --resume <id> || command -v xterm && xterm -e "qwen --resume <id>" || echo "No supported terminal found. Run manually: qwen --resume <id>"`
```

_— qwen3.6-plus via Qwen Code /review_
